pkgs/top-level/all-packages.nix

```diff
@@ -16792,6 +16792,13 @@ in
     inherit kernel;
     inherit (kernel) stdenv; # in particular, use the same compiler by default
 
+    autoModuleSignHook = kernel: makeSetupHook
+      { substitutions =
+        { kernel = kernel.dev + "/lib/modules/${kernel.modDirVersion}/build";
+          hash = if (kernel.configfile.structuredConfig ? MODULE_SIG_HASH)
```

Comment:
Another shortcoming, due to differing hash algos, we need to specify a default or read it in from the output config.